allauth/socialaccount/providers/quickbooks/provider.py

```diff
@@ -0,0 +1,60 @@
+from allauth.socialaccount import app_settings
+from allauth.socialaccount.providers.base import (
+    ProviderAccount,
+    ProviderException,
+)
+from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
+
+
+class QuickbooksAccount(ProviderAccount):
+
+    def to_str(self):
+        dflt = super(QuickbooksAccount, self).to_str()
+        name = self.account.extra_data.get('name', dflt)
+        first_name = self.account.extra_data.get('givenName', None)
+        last_name = self.account.extra_data.get('familyName', None)
+        if first_name and last_name:
+            name = first_name + ' ' + last_name
+        return name
+
+class QuickbooksOAuth2Provider(OAuth2Provider):
+    id = 'quickbooks'
+    # Name is displayed to ordinary users -- don't include protocol
+    name = 'Quickbooks'
```

Comment:
Sorry for nitpicking, can you name this one "QuickBooks" as that seems to be the official style. Please rename the classes as well (QuickbooksAccount -> QuickBooksAccount). Thanks.